chips/nrf52840/src/ieee802154_radio.rs

```diff
@@ -767,128 +835,254 @@ impl<'a> Radio<'a> {
+                if self.registers.event_end.is_set(Event::READY) {
+                    self.registers.event_end.write(Event::READY::CLEAR);
+                    let result = Ok(());
 
-        if self.registers.event_ccabusy.is_set(Event::READY) {
-            self.registers.event_ccabusy.write(Event::READY::CLEAR);
-            self.registers.event_ready.write(Event::READY::CLEAR);
-            self.registers.task_disable.write(Task::ENABLE::SET);
-            while self.registers.event_disabled.get() == 0 {}
-            self.registers.event_disabled.write(Event::READY::CLEAR);
-            //need to back off for a period of time outlined
-            //in the IEEE 802.15.4 standard (see Figure 69 in
-            //section 7.5.1.4 The CSMA-CA algorithm of the
-            //standard).
-            if self.cca_count.get() < IEEE802154_MAX_POLLING_ATTEMPTS {
-                self.cca_count.set(self.cca_count.get() + 1);
-                self.cca_be.set(self.cca_be.get() + 1);
-                let backoff_periods = self.random_nonce() & ((1 << self.cca_be.get()) - 1);
-                self.timer0
-                    .unwrap_or_panic() // Unwrap fail = Missing timer reference for CSMA
-                    .set_alarm(
-                        kernel::hil::time::Ticks32::from(0),
-                        kernel::hil::time::Ticks32::from(
-                            backoff_periods * (IEEE802154_BACKOFF_PERIOD as u32),
-                        ),
-                    );
-            } else {
-                self.transmitting.set(false);
-                //if we are transmitting, the CRCstatus check is always going to be an error
-                let result = Err(ErrorCode::BUSY);
-                //TODO: Acked is flagged as false until I get around to fixing it.
-                self.tx_client.map(|client| {
-                    let tbuf = self.tx_buf.take().unwrap(); // Unwrap fail = TX Buffer produced error when sending it back to the requestor after the channel was busy.
-                    client.send_done(tbuf, false, result)
-                });
+                    //TODO: Acked is hardcoded to always return false; add support to receive tx ACK
+                    self.tx_client.map(|client| {
+                        // Unwrap fail = TX Buffer is missing and was mistakenly not replaced after
+                        // completion of set_dma_ptr(...)
+                        let tbuf = self.tx_buf.take().unwrap();
+                        client.send_done(tbuf, false, result);
+                    });
+                    rx_init = true;
+                }
             }
+            RadioState::ACK => {
+                // Enforce READY_START shortcut enablement if radio is sending ACK
+                if !self.registers.shorts.is_set(Shortcut::READY_START) {
+                    panic!("READY_START shortcut not enabled while undergoing ACK.")
+                }
```

Comment:
Why enforce the shortcut here in the ACK state, but assume the shortcuts in the TX or RX states?

Reply:
All three originally were enforced. A comment from Amit prompted the switch to assuming rather than enforcing. I forgot to remove this one. Thanks for catching this.